packages/datagateway-common/src/detailsPanels/isis/datafileDetailsPanel.component.test.tsx

```diff
@@ -124,16 +177,28 @@ describe('Datafile details panel component', () => {
       },
     ];
 
-    const wrapper = createWrapper();
+    const mockDetailsPanelResize = jest.fn();
+    renderComponent({
+      rowData,
+      detailsPanelResize: mockDetailsPanelResize,
+    });
 
-    expect(detailsPanelResize).toHaveBeenCalledTimes(1);
+    await waitFor(() => {
+      expect(mockDetailsPanelResize).toHaveBeenCalledTimes(1);
+    });
 
-    wrapper.find('#datafile-parameters-tab').hostNodes().simulate('click');
+    await user.click(
+      await screen.findByRole('tab', {
+        name: 'datafiles.details.parameters.label',
+      })
+    );
 
-    expect(detailsPanelResize).toHaveBeenCalledTimes(2);
+    await waitFor(() => {
+      expect(mockDetailsPanelResize).toHaveBeenCalledTimes(2);
+    });
   });
 
-  it('does not call detailsPanelResize if not provided', () => {
+  it('should not call detailsPanelResize if not provided', async () => {
```

Comment:
I know `detailsPanelResize` is an optional parameter but do we ever run into a scenario where it's not provided? Because if not, we could make it a mandatory parameter and this test wouldn't even be needed (and maybe would require changing the `useLayoutEffect` hook we have as well).

Reply:
I actually don't know. I'll have a look to see if I can make it non-optional, but there has to be a reason why it is optional.

Reply:
Yeah because we have this test which is all well and good but it doesn't actually do anything 😅 like you instantiate a mock function, don't do anything with it and then assert that it's not called 😂 so yeah I think it would be good to see if it truly is non-optional and if so, this test can go